llvm/lib/Analysis/TargetLibraryInfo.cpp

```diff
@@ -550,6 +550,12 @@ static void initialize(TargetLibraryInfoImpl &TLI, const Triple &T,
     TLI.setUnavailable(LibFunc_nvvm_reflect);
   }
 
+  if (T.isBPF()) {
+    TLI.setUnavailable(LibFunc_rust_alloc);
```

Comment:
Why does llvm have to have any knowledge of these rust functions?

Reply:
it does. It replaces them with internal built-ins, which these calls to `TLI.setUnavailable()` prevent from happening.